src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedContainer.cs

```diff
@@ -123,16 +125,40 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
  
         public override ImmutableArray<Symbol> GetMembers()
         {
+            var builder = ArrayBuilder<Symbol>.GetInstance();
+            if (this.TypeKind == TypeKind.Struct)
+            {
+                // Add an extra object field for EnC.
+                builder.Add(GetStructFieldCaptureObject());
+            }
+
             Symbol constructor = this.Constructor;
-            return (object)constructor == null ? ImmutableArray<Symbol>.Empty : ImmutableArray.Create(constructor);
+            if ((object)constructor != null)
+            {
+                builder.Add(constructor);
+            }
+
+            return builder.ToImmutableAndFree();
         }
 
         public override ImmutableArray<Symbol> GetMembers(string name)
         {
+            if (this.TypeKind == TypeKind.Struct && name == StructFieldCaptureName)
+            {
+                // Add an extra object field for EnC.
```

Comment:
>// Add an extra object field for E [](start = 16, length = 34)

We should only add the field to synthesized containers that implement local function closures, not to async state machines. 

Also this field should only be added in debug builds.